src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java

```diff
@@ -36,27 +44,36 @@ public void processContributionsForStages(Set<Stage> stages) {
             return;
         }
 
-        contributionSomuItems = filterContributions(contributionSomuItems);
-
-        if (contributionSomuItems.size() == 0) {
-            return;
-        }
-
         for (Stage stage :
                 stages) {
             if (MPAMContributionStages.contains(stage.getStageType())
                     || FOIContributionStages.contains(stage.getStageType())
                     || COMPLIANT_CASE_TYPE.equals(stage.getCaseDataType())) {
-
-                Set<SomuItem> caseContributions = filterContributionsByCase(stage.getCaseUUID(), contributionSomuItems);
-
-                calculateDueContributionDate(caseContributions)
+                Set<Contribution> contributions =
+                        contributionSomuItems.stream()
+                                .filter(somuItem -> somuItem.getCaseUuid().equals(stage.getCaseUUID()))
+                                .map(somuItem -> {
+                                    try {
+                                        return objectMapper.readValue(somuItem.getData(), Contribution.class);
+                                    } catch (JsonProcessingException e) {
+                                        log.error(String.format("Failed to process somu item for reason: %s", e.getMessage()), e);
+                                    }
+                                    return null;
+                                })
+                                .filter(Objects::nonNull)
+                                .filter(Contribution::isContribution).collect(Collectors.toSet());
+
+                if (contributions.size() == 0) {
+                    continue;
+                }
+
+                calculateDueContributionDate(contributions)
                         .ifPresent(ld -> {
                             log.info("Setting contribution date {}, for caseId {}", ld, stage.getCaseUUID());
                             stage.setDueContribution(ld.toString());
                         });
 
-                highestContributionStatus(caseContributions)
+                highestContributionStatus(contributions)
                         .ifPresent(cs -> {
```

Comment:
Will NONE count as present here?

Reply:
Yeah it would, changed to filter out NONE in the furthest inner function...